### PR TITLE
Network dashboard - fixing 2 bugs: overriden variable and page token …

### PR DESCRIPTION
**Pull request by aurelienlegrand**
Network Quota monitoring: Fixing 2 bugs: overriden variable and page token not handled properly

### CODE REVIEW

File: blueprints/cloud-operations/network-quota-monitoring/src/plugins/utils.py [hunk 1]
```diff
@@ -65,8 +65,10 @@ def parse_page_token(data, url):
   page_token = data.get('nextPageToken')
   if page_token:
     logging.info(f'page  token {page_token}')
-  if page_token:
-    return RE_URL.sub(f'pageToken={page_token}&', url)
+    if 'pageToken' in url:
+      return RE_URL.sub(f'pageToken={page_token}', url)
+    else:
+      return f'{url}&pageToken={page_token}'
```
Comment (wiktorn) [inline blueprints/cloud-operations/network-quota-monitoring/src/plugins/utils.py:71]:
```suggestion
    parsed_url = urllib.parse.urlparse(url)
    parsed_qs = urllib.parse.parse_qs(parsed_url.query)
    parsed_qs['pageToken'] = page_token
    return parsed_url._replace(query=urllib.parse.urlencode(parsed_qs, doseq=True)).geturl()
```

And then you can get rid of `RE_URL`

**Review by wiktorn — APPROVED** [review]

File: blueprints/cloud-operations/network-quota-monitoring/src/plugins/core-discover-cai-nodes.py [hunk 2]
```diff
@@ -43,9 +43,9 @@ def _handle_discovery(resources, response, data):
       yield Resource('folders', name, {'name': result['displayName']})
     elif asset_type == 'Project':
       number = result['project'].split('/')[1]
-      data = {'number': number, 'project_id': name}
-      yield Resource('projects', name, data)
-      yield Resource('projects:number', number, data)
+      data_proj = {'number': number, 'project_id': name}
```
Comment (ludoo) [inline blueprints/cloud-operations/network-quota-monitoring/src/plugins/core-discover-cai-nodes.py:46]: there's no need to rename `data`

File: blueprints/cloud-operations/network-quota-monitoring/src/plugins/utils.py (hunk 1 above)
Comment (ludoo) [reply blueprints/cloud-operations/network-quota-monitoring/src/plugins/utils.py:71]: I find yours harder to read Wiktor :)

File: blueprints/cloud-operations/network-quota-monitoring/src/plugins/core-discover-cai-nodes.py (hunk 2 above)
Comment (ludoo) [reply blueprints/cloud-operations/network-quota-monitoring/src/plugins/core-discover-cai-nodes.py:46]: can you expand on what the bug is here?

Comment (wiktorn) [reply blueprints/cloud-operations/network-quota-monitoring/src/plugins/core-discover-cai-nodes.py:46]: It hides the function input parameter `data`, which I imagine is needed in L51 - `parse_page_token` will look for `pageToken` there and fail...

File: blueprints/cloud-operations/network-quota-monitoring/src/plugins/utils.py (hunk 1 above)
Comment (wiktorn) [reply blueprints/cloud-operations/network-quota-monitoring/src/plugins/utils.py:71]: Yes, it is a bit more complex, but handles all the edge cases.

`RE_URL.sub` will work for now, so I'm open to reject my suggestion.

**Review by wiktorn — APPROVED** [review]

File: blueprints/cloud-operations/network-quota-monitoring/src/plugins/core-discover-cai-nodes.py (hunk 2 above)
Comment (ludoo) [reply blueprints/cloud-operations/network-quota-monitoring/src/plugins/core-discover-cai-nodes.py:46]: ah yes sorry, data is used in the rest of the function...

File: blueprints/cloud-operations/network-quota-monitoring/src/plugins/utils.py [hunk 3]
```diff
@@ -34,7 +34,7 @@
 Host: compute.googleapis.com
 
 '''
-RE_URL = re.compile(r'nextPageToken=[^&]+&?')
+RE_URL = re.compile(r'pageToken=[^&]+&?')
```
Comment (aurelienlegrand) [inline blueprints/cloud-operations/network-quota-monitoring/src/plugins/utils.py:37]: I forgot one last change here, in the body of the HTTP reply it is called 'nextPageToken'  but in the URL it should be 'pageToken'.

**Pull request merged**